qiskit/transpiler/passes/synthesis/unitary_synthesis.py

```diff
@@ -137,99 +142,200 @@ def run(self, dag: DAGCircuit) -> DAGCircuit:
             Output dag with UnitaryGates synthesized to target basis.
 
         Raises:
-            TranspilerError:
-                1. pulse_optimize is True but pulse optimal decomposition is not known
-                   for requested basis.
-                2. pulse_optimize is True and natural_direction is True but a preferred
-                   gate direction can't be determined from the coupling map or the
-                   relative gate lengths.
+            QiskitError: if a 'method' was specified for the class and is not
+                found in the installed plugins list. The list of installed
+                plugins can be queried with
+                :func:`~qiskit.transpiler.passes.synthesis.plugins.unitary_synthesis_plugin_names`
         """
+        if not self.method:
+            method = "default"
+        else:
+            method = self.method
+        if method not in self.plugins.ext_plugins:
+            raise QiskitError("Specified method: %s not found in plugin list" % method)
+        plugin_method = self.plugins.ext_plugins[method].obj
+        if plugin_method.supports_coupling_map:
+            dag_bit_indices = {bit: idx for idx, bit in enumerate(dag.qubits)}
+        for node in dag.named_nodes(*self._synth_gates):
+            synth_dag = None
+            kwargs = {}
+            if plugin_method.supports_basis_gates:
+                kwargs["basis_gates"] = self._basis_gates
+            if plugin_method.supports_coupling_map:
+                kwargs["coupling_map"] = self._coupling_map
+                kwargs["qubits"] = [dag_bit_indices[x] for x in node.qargs]
```

Comment:
This adds an extra argument `qubits` to the kwargs conditioned on `plugin_method.supports_coupling_map`, but I don't remember reading any documentation about that argument?  (It's entirely possible I missed it, though, so if it is there, apologies.)